sky/clouds/aws.py

```diff
@@ -695,6 +691,48 @@ def _get_disk_specs(cls, disk_tier: Optional[str]) -> Dict[str, Any]:
             'custom_disk_perf': tier != 'low',
         }
 
+    @classmethod
+    def check_quota_available(cls,
+                              region: str,
+                              instance_type: str,
+                              use_spot: bool = False) -> bool:
+        """
+        Check if the quota is available for the requested instance_type
+
+        AWS-specific implmentation of check_quota_available. The function works by
+        matching the instance_type to the corresponding AWS quota code, and then using
+        the boto3 Python API to query the region for the specific quota code.
+
+        Returns:
+            False if the quota is found to be zero, and True otherwise.
+        Raises:
+            ImportError: if the dependencies for AWS are not able to be installed.
+            botocore.exceptions.ClientError: error in Boto3 client request.
+        """
+
+        from sky.clouds.service_catalog import aws_catalog  # pylint: disable=import-outside-toplevel,unused-import
+
+        quota_code = aws_catalog.get_quota_code(instance_type, use_spot)
+
+        if quota_code is None:
+            # Quota code not found in the catalog for the chosen instance_type, try provisioning anyway
+            return True
+
+        client = boto3.client('service-quotas', region_name=region)
```

Comment:
Our adaptor will handle the import.
```suggestion
        client = aws.client('service-quotas', region_name=region)
```